set-client/src/main/java/eu/europeana/set/client/connection/BaseApiConnection.java

```diff
@@ -1,156 +1,340 @@
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug(ERROR_MESSAGE + " {} ", errorResponse.getMessage());
+                }
+                throw new SetApiClientException(ERROR_MESSAGE + errorResponse.getMessage(), response.getCode());
+            }
+            return String.valueOf(response.getCode());
+        } catch (IOException | ParseException e) {
+            throw new SetApiClientException(ERROR_MESSAGE + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR, e);
+        }
+    }
+
+    private UserSet parseSetApiResponse(CloseableHttpResponse response, List<Integer> statusToCheckList) throws SetApiClientException, IOException, ParseException {
+        String responseBody = EntityUtils.toString(response.getEntity());
+        if (statusToCheckList.contains(response.getCode())) {
+            return mapper.readValue(responseBody, BaseUserSet.class);
+        } else {
+            AbstractUserSetApiResponse errorResponse = mapper.readValue(responseBody, AbstractUserSetApiResponse.class);
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(ERROR_MESSAGE + " {} ", errorResponse.getMessage());
+            }
+            throw new SetApiClientException(ERROR_MESSAGE + errorResponse.getMessage(), response.getCode());
+        }
+
+    }
+
+    /**
+     * Fetches the get user Set pagination response
+     *
+     * Check profile for the deserialization
+     * META - empty page (no items) ,  ITEMS_META - only set descriptions, not item descriptions , ITEMS - items set as set ids
+     * ITEMS is the default profile value
+     * @param url
+     * @param authorizationHeaderValue
+     * @return
+     * @throws SetApiClientException
+     */
+    protected List<RecordPreview> getUserSetPaginatedResponse(String url, String authorizationHeaderValue, String profile) throws SetApiClientException {
+        try {
+            LOGGER.trace("Call to Get UserSet API (Paginated): {} ", url);
+            CloseableHttpResponse response = getHttpConnection().get(url, ContentType.APPLICATION_JSON.getMimeType(), authorizationHeaderValue);
+            String responseBody = EntityUtils.toString(response.getEntity());
+            if (response.getCode() == HttpStatus.SC_OK) {
+                if (StringUtils.equals(profile, ProfileConstants.VALUE_PARAM_ITEMS)) {
+                    TypeReference<ResultsPageImpl<String>> typeRef = new TypeReference<>() {};
+                    List<String> recordIds  = mapper.readValue(responseBody, typeRef).getItems();
+                    List<RecordPreview> records = new ArrayList<>();
+                    for (String id: recordIds) {
+                       records.add(new RecordPreview(id));
+                    }
+                    return records;
+                }
+                TypeReference<ResultsPageImpl<RecordPreview>> typeRef = new TypeReference<>() {};
```

Comment:
we should have an if/else or switch for profiles. I assume this call is parsing the items.meta profile. see https://docs.google.com/spreadsheets/d/19_5ihd_MKuVenIno-enankAu8bG7v79TGHQuepOtTv8/edit?gid=1005516996#gid=1005516996

Reply:
set API will return an Invalid profile Exception if there is a wrong parameter passed. In everycase we return List<RecordPreview>. But for the profile where only item ids are returned, we need to do bit more processing. here for "items" profile we get List<String> which is converted to List<RecordPreview> with id value only

Reply:
I have added a deserializer now to handle all the profile responses